src/etc/inc/filter.inc

```diff
@@ -2048,7 +2048,7 @@ function filter_generate_address(&$FilterIflist, &$rule, $target = 'source', $is
         $matches = "";
         if ($network_name == '(self)') {
             $src = "(self)";
-        } elseif (preg_match("/(.*)ip$/", $rule[$target]['network'], $matches)) {
+        } elseif (preg_match("/(^[wan|lan|opt]+[0-9]*)ip$/", $rule[$target]['network'], $matches)) {
```

Comment:
@fabianfrz thanks, that makes sense when you put it this way :) I'll fix the commit soon.